### PR TITLE
support additional configuration for the upstream LSP

### PR DESCRIPTION
**Pull request by phryneas**

### CODE REVIEW

File: src/language-server/project/rover/project.ts [hunk 1]
```diff
@@ -150,7 +150,18 @@ export class RoverProject extends GraphQLProject {
         "Connection is closed.",
       );
     }
-    const child = cp.spawn(this.config.rover.bin, ["lsp"], {
+    const args = ["lsp", "--elv2-license", "accept"];
```
Comment (phryneas) [inline src/language-server/project/rover/project.ts:161]: @dylan-apollo could you please verify that I understood you correctly and this should always be added? :)

File: src/language-server/config/config.ts [hunk 2]
```diff
@@ -104,6 +106,21 @@ const roverConfig = z.object({
...
+      return existsSync(supergraphConfig) ? supergraphConfig : undefined;
+    }, z.string().nullable().optional())
+    .describe(
+      "The path to your `supergraph.yml` file. \n" +
+        "Defaults to a `supergraph.yml` in the folder of your `apollo.config.js`, if there is one.",
+    ),
+  extraArgs: z
+    .array(z.string())
+    .default([])
+    .describe("Extra arguments to pass to the Rover CLI."),
```
Comment (phryneas) [inline src/language-server/config/config.ts:123]: I believe these should be enough for now, and `extraArgs` opens the door for anything I might have forgotten for now :)

File: src/language-server/config/config.ts [hunk 3]
```diff
@@ -29,8 +30,9 @@ function ignoredFieldWarning(
 export interface Context {
   apiKey?: string;
   serviceName?: string;
+  configPath?: string;
 }
-const context = new Slot<Context>();
+const contextStore = new AsyncLocalStorage<Context>();
```
Comment (phryneas) [inline src/language-server/config/config.ts:35]: Switching to `AsyncLocalStorage` here for consistency - I need to use that in other places already.

File: src/language-server/project/rover/project.ts (hunk 1 above)
Comment (dylan-apollo) [reply src/language-server/project/rover/project.ts:161]: Yep!

**Review by dylan-apollo — APPROVED** [review]
Looks good to me!

**Review by jerelmiller — APPROVED** [review]
🎉

**Pull request merged**